src/cauchy.jl

```diff
@@ -2,28 +2,27 @@
 
 export Cauchy
 
-immutable Cauchy{T} <: AbstractMatrix{T}
-    x::Vector{T} #
-    y::Vector{T} #
-end # immutable
-
-function Cauchy(k::Number)
-         Cauchy([1:k],[1:k])
+immutable Cauchy{T<:Number} <: AbstractMatrix{T}
+    x::Vector{T}
+    y::Vector{T}
 end
 
-function Cauchy(x::Vector)
-         Cauchy(x,x)
+function Cauchy(x, y)
+    T = promote_type(eltype(x), eltype(y))
+    vx = Vector{T}(x)
+    vy = Vector{T}(y)
+    Cauchy(vx, vy)
 end
+function Cauchy(x)
+    vx = Vector(x)
+    Cauchy(vx, vx)
+end
+Cauchy(k::Number) =  Cauchy(1:k)
 
-# Define its size
-
-size(A::Cauchy, dim::Integer) = length(A.x)
```

Comment:
This method is still needed for `size(A, 1)` to work.

Reply:
That might have been needed in 2017 but at least as of today with v1.6 it is no longer needed.
Base has a default for it now.  (I don't know when that default was added...)